packages/core/src/agent/Agent.ts

```diff
@@ -171,14 +172,45 @@ export class Agent {
     // Because this requires the connections module, we do this in the agent constructor
     if (mediatorConnectionsInvite) {
       // Assumption: processInvitation is a URL-encoded invitation
-      let connectionRecord = await this.connections.receiveInvitationFromUrl(mediatorConnectionsInvite, {
-        autoAcceptConnection: true,
-      })
-
-      // TODO: add timeout to returnWhenIsConnected
-      connectionRecord = await this.connections.returnWhenIsConnected(connectionRecord.id)
-      const mediationRecord = await this.mediationRecipient.requestAndAwaitGrant(connectionRecord, 60000) // TODO: put timeout as a config parameter
-      await this.mediationRecipient.setDefaultMediator(mediationRecord)
+      const invitation = await ConnectionInvitationMessage.fromUrl(mediatorConnectionsInvite)
+      // Check if invitation has been used already
+      const connections = await this.connections.getAll()
+      let defaultMediatorBootstrapped = false
+      for (const connection of connections) {
+        if (connection.invitation === invitation) {
+          this.logger.warn(
+            `Mediator Invitation in configuration has already been used to ${
+              connection.isReady ? 'make' : 'initialized'
+            } a connection`
+          )
+          if (connection.isReady) {
+            const mediators = await this.mediationRecipient.getMediators()
```

Comment:
This way we don't have to retrieve all mediators:

```suggestion
            const mediator = await this.mediationRecipient.findByConnectionId(connection.id)
```